### ISSUE TITLE
mutlithreading

### ISSUE THREAD

**Issue by koreyr** [maltesie/bridge2#1]
Thank you for the excellent work on this application it is highly useful. When I compute networks including nearby waters bridge2 only uses a single core to compute the results. I am not well verse in Qt applications or multithreading but I looked through the multithreading.py folder and am curious if there is anyway to increase the number of cores used while performing the analysis. 

Is there something I need to do to run the analysis on more than one core, e.g. env variable?

**Comment by maltesie** [comment]
I'm happy you find it useful. The multithreading right now only decouples the interface from the computations. I will check how easy it is to implement multithreading for the H-bond computations and get back to you. For water wires it will be difficult, since the computations in each frame depend on the frame before, but for H-bonds it should be simple enough.

**Comment by koreyr** [comment]
Thank you for the reply. That would be wonderful, thank you. By any chance is there a standard CLI interface so I can batch script the analysis to produce many baf files at once and load them into the graphical interface? Would this provide any benefit?

**Comment by maltesie** [comment]
I will try to implement the multithreading over the weekend. There is no CLI to bridge yet, but batch computations make a lot of sense. Adding a whole CLI will be too much work for me in the near future. I could think of a workaround where you open a text file with a list of structure file names and one with the corresponding trajectories. Would that help? If you have other suggestions, I'm happy to hear them :)

**Comment by koreyr** [comment]
That would be great,  especially if it could handle spawning each
trajectory calculation on a unique core,  or multithreading each trajectory
analysis for hbonds,  whichever is easier.

**Comment by maltesie** [comment]
Hi koreyr,

I added batch computations and multithreading for the centrality computations. I hope I can add multithreading for the HBond computations soon, but that should already speed things up. 

To run batches, list all structure files in a text file and all corresponding trajectory files (separated by comma, if several per job) into another text file and then select those files in the "New Analysis" dialog.

To set the number of threads to use, change the last line in the settings.ini file (I will add something to the GUI soon)

I tested the new additions and everything seems to work as expected, but I'd be happy to know if something is off with your computations.

**Comment by maltesie** [comment]
Hi @koreyr

I added multithreading to the HBond computations (not for water wires yet) and fixed some bugs that came with the batch computations and the multithreading of the centrality computations. All in all, the speedup is not good (around 20%, independent of the number of threads). I suspect the DCD reader to be the bottleneck, but have to investigate further. I reduced the number frames, for which position information is saved to 100 max, which also speeds up the overall process a lot.

I hope this is useful so far. 

Best,
Malte
